### PR TITLE
Add the `rpy2-arrow` package.

### PR DESCRIPTION
**Pull request by matthewwardrop**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewwardrop** [comment]
@conda-forge/help-python, ready for review!

cc: @lgautier

**Comment by matthewwardrop** [comment]
@conda-forge/help-r The R interpreter is not being properly found during testing. How do you get the R environment correctly set up during build?

**Comment by matthewwardrop** [comment]
@conda-forge/help-r Bump?

**Comment by matthewwardrop** [comment]
@conda-forge/help-python ready to merge. I disabled the test on Windows so as not to block things unnecessarily. There is no windows specific code here, so things will very likely work if they work on Linux and Mac.

**Comment by matthewwardrop** [comment]
@BastianZim Are you able to review this PR? Or do you know who I should ask?

**Comment by BastianZim** [comment]
Unfortunately not as it involves R but let me ping the team for you. @conda-forge/help-r

**Comment by matthewwardrop** [comment]
@BastianZim Just in case it wasn't clear, this is a Python package; it just sits at the boundary between Python and R allowing fast passing of Arrow data structures between Python and R interpreters running in the same process.

File: recipes/rpy2-arrow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+  run:
+    - python >=3.6
+    - rpy2 >=3.5
+    - pyarrow >=5
+    - r-arrow >=5
+
+test:
+  imports:
+    - rpy2_arrow.pyarrow_rarrow  # [not win]  # R environment in conda build not correctly configured, but will be at runtime by the user.
```
Comment (BastianZim) [inline recipes/rpy2-arrow/meta.yaml:29]:
```suggestion
    - rpy2_arrow.pyarrow_rarrow
```
You can remove this. The feedstock will only build in linux.

**Review by BastianZim — CHANGES_REQUESTED** [review]
> Just in case it wasn't clear, this is a Python package; it just sits at the boundary between Python and R allowing fast passing of Arrow data structures between Python and R interpreters running in the same process.

Ahh I understand. Have added one comment. It's fine if the test fails since this will only be built on Linux. Once you remove that I can merge.

File: recipes/rpy2-arrow/meta.yaml (hunk 1 above)
Comment (matthewwardrop) [reply recipes/rpy2-arrow/meta.yaml:29]: Oh, interesting. I added that in order to get the builds to pass in the staging repo. I'll just remove the commit.

**Comment by matthewwardrop** [comment]
@BastianZim Should be ready now!

**Comment by matthewwardrop** [comment]
Thank you @BastianZim !

**Comment by lgautier** [comment]
Thanks to all!

If something breaks down line, and it should likely be fixed upstream (in the package `rpy2-arrow`) do not hesitate to open an issue here: https://github.com/rpy2/rpy2-arrow

**Pull request merged**